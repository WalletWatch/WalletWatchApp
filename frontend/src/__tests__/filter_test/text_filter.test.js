import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';
import TextFilter from '../../components/aggrid_filter/filters/text_filter';

// Mocking the grid API
const mockApi = {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    isDestroyed: jest.fn().mockReturnValue(false),
    forEachNode: jest.fn(),
    getColumnDefs: jest.fn(),
    getAllDisplayedColumns: jest.fn(),
    setColumnFilterModel: jest.fn().mockResolvedValue(true), // Mocking a successful promise
    onFilterChanged: jest.fn(),
    };

    describe('TextFilter', () => {
    it('renders correctly', () => {
        const { container } = render(<TextFilter api={mockApi} node="test" />);
        expect(container).toMatchSnapshot();
    });

    it('applies filter on button click', async () => {
        // Mocking the values returned by forEachNode
        mockApi.forEachNode.mockImplementationOnce(callback => {
            callback({ displayed: true, data: { test: 'value1' } });
            callback({ displayed: true, data: { test: 'value2' } });
            callback({ displayed: true, data: { test: 'value3' } });
        });

        const { getByText, getByPlaceholderText } = render(<TextFilter api={mockApi} node="test" />);

        // Triggering the filter application
        // fireEvent.click(getByText('Apply'));

        // Expectations
        // expect(mockApi.setColumnFilterModel).toHaveBeenCalledWith('test', { values: ['value1', 'value2', 'value3'] });
        // expect(mockApi.onFilterChanged).toHaveBeenCalled(0);
    });

    // it('resets filter on button click', async () => {
    //     // Mocking the values returned by forEachNode
    //     mockApi.forEachNode.mockImplementationOnce(callback => {
    //     callback({ displayed: true, data: { test: 'value1' } });
    //     callback({ displayed: true, data: { test: 'value2' } });
    //     callback({ displayed: true, data: { test: 'value3' } });
    //     });

    //     const { getByText } = render(<TextFilter api={mockApi} node="test" />);

    //     // Triggering the filter reset
    //     fireEvent.click(getByText('Reset'));

    //     // Expectations
    //     expect(mockApi.setColumnFilterModel).toHaveBeenCalledWith('test', { values: ['value1', 'value2', 'value3'] });
    //     expect(mockApi.onFilterChanged).toHaveBeenCalled();
    // });
});
