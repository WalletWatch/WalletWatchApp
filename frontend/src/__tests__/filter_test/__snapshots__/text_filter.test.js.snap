// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFilter renders correctly 1`] = `
<div>
  <div
    class="column-filter"
    id="test"
  >
    <div
      style="overflow: hidden; height: 150px;"
    >
      <input
        class="title"
        id="searchertest"
        placeholder="Search for test..."
        type="text"
        value=""
      />
      <div
        class="column-filter-wrapper"
        id="searchtest"
        style="height: calc(100% - 28px);"
      >
        <div
          class="column-filter-item"
        >
          <input
            checked=""
            class="input-cols-filter"
            id="AllParamstest"
            type="checkbox"
            value="true"
          />
          <label
            class="input-column-name"
            for="AllParamstest"
          >
            (All)
          </label>
        </div>
      </div>
    </div>
    <div
      class="apply-button-wrapper"
    >
      <button
        class="apply-button"
      >
        Reset
      </button>
      <button
        class="apply-button"
        style="margin-left: 5px;"
      >
        Apply
      </button>
    </div>
  </div>
</div>
`;
